scripts/testing/tolerance_test.py

```diff
@@ -37,32 +37,54 @@ def ensure_file(path):
         print("ERROR: Given file is not a file: {0}".format(path))
         sys.exit(1)
 
-
 def parse_args():
     parser = argparse.ArgumentParser(description="Compare two Serac summary files")
+    group = parser.add_mutually_exclusive_group(required=True)
 
     parser.add_argument("--baseline", type=str, required=True,
                         help="Path to baseline summary file")
     parser.add_argument("--test", type=str, required=True,
                         help="Path to test summary file")
-    parser.add_argument("--tolerance", type=float, required=True,
+    group.add_argument("--tolerance", type=str,
                         help="Allowed tolerance amount for individual values")
+    group.add_argument("--tolerance-file", type=str,
+                        help="JSON file specifying tolerance amount for specific values")
 
     args = parser.parse_args()
 
     # Ensure correctness of given options
     ensure_file(args.baseline)
     ensure_file(args.test)
+    if args.tolerance_file is not None:
+        ensure_file(args.tolerance_file)
 
     # Print options
     print("------- Given Options -------")
-    print("Baseline file: {0}".format(args.baseline))
-    print("Test file:     {0}".format(args.test))
-    print("Tolerance:     {0}".format(args.tolerance))
+    print("Baseline file:  {0}".format(args.baseline))
+    print("Test file:      {0}".format(args.test))
+    print("Tolerance:      {0}".format(args.tolerance))
+    print("Tolerance file: {0}".format(args.tolerance_file))
     print("-----------------------------")
 
     return args
 
+# Takes query string and returns a tolerance dictionary
+def parse_tolerance_query(query):
+    tolerance_dict = {}
+
+    # Two cases:
+    # 1. single value (0.1)
+    # 2. specific value(s) (velocity:0.1::displacement:0.025)
+
+    if ":" not in query:
+        tolerance_dict = { "default": float(query) }
+    else:
+        query_list = query.split("::")
```

Comment:
@jamiebramwell you originally suggested the delimiter `::`.  Are you okay with a different delimiter to make the options stand out more, possibly `,`?

`--tolerances=velocity:0.1::displacement:0.234`

vs

--tolerances=velocity:0.1,displacement:0.234